pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -498,6 +498,12 @@ func (c *Cluster) provisionOSDContainer(devices []rookalpha.Device, selection ro
 				logger.Infof("%s osds requested on device %s (node %s)", count, device.Name, nodeName)
 			}
 			devSuffix += ":" + count
+			if deviceClass, ok := device.Config[config.DeviceClassKey]; ok {
+				logger.Infof("osd %s requested with deviceClass %s (node %s)", device.Name, deviceClass, nodeName)
+				devSuffix += ":" + deviceClass
+			} else {
+				devSuffix += ":auto"
```

Comment:
Instead of ":auto" how about just ":"? A blank device class would imply it's automatically handled.